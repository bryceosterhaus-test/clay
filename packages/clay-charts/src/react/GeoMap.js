import React from 'react';
import {Geomap} from '../util/geo-map';

export default class GeomapReact extends React.Component {
	constructor(props) {
		super(props);

		this._containerRef = React.createRef();
	}
	componentDidMount() {
		this._geoMapInstance = new Geomap({
			...this.props,
			element: this._containerRef.current,
		});

		this._geoMapInstance.attached();
	}

	/**
	 * @inheritDoc
	 */
	componentWillUnmount() {
		this._geoMapInstance.disposed();
	}

	render() {
		const {data, element, color, pollingInterval, ...otherProps} = this.props;

		const {height = '100%', width = '100%'} = this._geoMapInstance
			? this._geoMapInstance.getSize()
			: {};

		return (
			<div
				style={{height, width}}
				{...otherProps}
				ref={this._containerRef}
			/>
		);
	}
}
