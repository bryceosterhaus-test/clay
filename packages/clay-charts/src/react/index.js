import React from 'react';
import BillboardChart from 'react-billboardjs';
import GeoMap from './react/GeoMap';

import {
	DEFAULT_COLORS,
	DEFAULT_GRID_OBJECT,
	DEFAULT_LINE_CLASSES,
	DEFAULT_POINT_PATTERNS,
} from '../ChartBase';

import '../scss/main.scss';

export default class extends React.Component {
	render() {
		const {color, data, grid, line, point, ...otherProps} = this.props;

		let ChartComponent;
		let props = {data, ...otherProps};

		switch (data.type) {
			case 'geo-map':
				ChartComponent = GeoMap;
				break;
			default:
				ChartComponent = BillboardChart;
		}

		return (
			<ChartComponent
				{...otherProps}
				data={data}
				grid={Object.assign(DEFAULT_GRID_OBJECT, grid)}
				color={Object.assign({pattern: DEFAULT_COLORS}, color)}
				line={Object.assign({classes: DEFAULT_LINE_CLASSES}, line)}
				point={Object.assign(
					{
						pattern: DEFAULT_POINT_PATTERNS,
					},
					point
				)}
			/>
		);
	}
}
