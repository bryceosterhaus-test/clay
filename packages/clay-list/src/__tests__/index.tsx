/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ClayList from '..';

describe('ClayList', () => {
	it('renders', () => {
		const testRenderer = TestRenderer.create(
			<ClayList />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});
