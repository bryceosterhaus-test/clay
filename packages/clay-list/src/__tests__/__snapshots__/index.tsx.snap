// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayList renders 1`] = `
<ul
  className="list-group show-quick-actions-on-hover"
>
  <li
    className="list-group-header"
  >
    <h3
      className="list-group-header-title"
    >
      This is a header
    </h3>
  </li>
  <li
    className="list-group-item list-group-item-flex"
  >
    <div
      className="autofit-col"
    >
      Item 1
    </div>
    <div
      className="autofit-col"
    >
      ItemField
    </div>
    <div
      className="autofit-col autofit-col-expand"
    >
      <h4
        className="list-group-title"
      >
        Item Title and expanded
      </h4>
      <p
        className="list-group-text"
      >
        Item Text
      </p>
    </div>
    <div
      className="autofit-col"
    >
      ItemField
    </div>
  </li>
  <li
    className="list-group-item list-group-item-flex"
  >
    <div
      className="autofit-col"
    >
      Item 2
    </div>
    <div
      className="autofit-col autofit-col-expand"
    >
      Hover this item for action menu
    </div>
    <div
      className="autofit-col"
    >
      <div
        className="quick-action-menu"
      >
        <button
          className="component-action quick-action-item"
          onClick={[Function]}
          role="button"
        >
          <svg
            className="lexicon-icon lexicon-icon-trash"
            role="presentation"
          >
            <use
              xlinkHref="/foo/bar#trash"
            />
          </svg>
        </button>
        <button
          className="component-action quick-action-item"
          onClick={[Function]}
          role="button"
        >
          <svg
            className="lexicon-icon lexicon-icon-cog"
            role="presentation"
          >
            <use
              xlinkHref="/foo/bar#cog"
            />
          </svg>
        </button>
      </div>
    </div>
  </li>
  <li
    className="list-group-header"
  >
    <h3
      className="list-group-header-title"
    >
      This is another header
    </h3>
  </li>
  <li
    className="list-group-item list-group-item-flex"
  >
    <div
      className="autofit-col"
    >
      Item 3
    </div>
    <div
      className="autofit-col"
    >
      ItemField
    </div>
  </li>
</ul>
`;
